internal/pod_models.go

```diff
@@ -733,7 +733,8 @@ func usePvc(cluster *fdbv1beta2.FoundationDBCluster, processClass fdbv1beta2.Pro
 			storage = &storageCopy
 		}
 	}
-	return processClass.IsStateful() && (storage == nil || !storage.IsZero())
+	// TODO can a stateful process ever have this be true? maybe with testing?
+	return storage == nil || !storage.IsZero()
```

Comment:
I believe this could be the case if someone doesn't specify a `volumeClaimTemplate` in the `FoundationDBCluster` spec, e.g. if this part: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/config/tests/base/cluster.yaml#L42-L47 would be missing.